src/session/session_api.c

```diff
@@ -2219,6 +2219,38 @@ __session_get_last_error(WT_SESSION *wt_session, int *err, int *sub_level_err, c
     *err_msg = session->err_info.err_msg;
 }
 
+/*
+ * __session_set_last_error --
+ *     WT_SESSION->set_last_error method.
+ */
+static int
+__session_set_last_error(WT_SESSION *wt_session, int err, int sub_level_err)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    const char *msg_content;
+    size_t msg_size;
+
+    /* Free the last error message string, if it was allocated. */
+    if (err_msg != NULL) {
+        __wt_free(session, err_msg);
+    }
```

Comment:
We usually don't add curly brackets for one-liners.
Furthermore, should we simply call `__wt_free(session, err_msg);` all the time? It should be a no-op if `NULL`.

Reply:
Ah I see, I forgot `__wt_free` handles it automatically.

Reply:
Fixed in [Added assert to check for valid sub_level_err, formatting changes](https://github.com/wiredtiger/wiredtiger/pull/11350/commits/f0f443ad444cf6fe5f2fa02603569cd679c53dd6)